crates/router/src/db/payment_attempt.rs

```diff
@@ -465,33 +390,39 @@ mod storage {
                         .change_context(errors::StorageError::KVError)?;
                     let field = format!("pa_{}", updated_attempt.attempt_id);
                     let updated_attempt = self
-                        .redis_conn()
-                        .map_err(Into::<errors::StorageError>::into)?
+                        .redis_conn
                         .set_hash_fields(&key, (&field, &redis_value))
                         .await
                         .map(|_| updated_attempt)
                         .change_context(errors::StorageError::KVError)?;
 
-                    let conn = connection::pg_connection_write(self).await?;
-                    // Reverse lookup for connector_transaction_id
-                    if let (None, Some(connector_transaction_id)) = (
+                    match (
                         old_connector_transaction_id,
                         &updated_attempt.connector_transaction_id,
                     ) {
-                        let field = format!("pa_{}", updated_attempt.attempt_id);
-                        ReverseLookupNew {
-                            lookup_id: format!(
-                                "{}_{}",
-                                &updated_attempt.merchant_id, connector_transaction_id
-                            ),
-                            pk_id: key.clone(),
-                            sk_id: field.clone(),
-                            source: "payment_attempt".to_string(),
+                        (None, Some(connector_transaction_id)) => {
+                            add_connector_txn_id_to_reverse_lookup(
```

Comment:
Here key is merchant_id_payment_id. But while creating key initially in insert they key is merchant_id_attempt_id.